tests/test_base_device.py

```diff
@@ -141,29 +138,30 @@ class TestSupportForV1andV2:
 
     @pytest.mark.parametrize(
         "backend",
-        [
-            legacy_backend,
-            backend,
-        ],
+        [legacy_backend, backend, mocked_backend],
     )
     def test_v1_and_v2_mocked(self, backend):
         """Test that device initializes with no error mocked"""
         dev = QiskitDevice2(wires=10, backend=backend, use_primitives=True)
         assert dev._backend == backend
 
-    @pytest.mark.skip(
-        reason="Fake backends do not have attribute _service, should address in (SC 55725)"
-    )
     @pytest.mark.parametrize(
-        "backend",
+        "backend, use_primitives, shape",
         [
-            FakeManila(),
-            FakeManilaV2(),
+            (FakeManila(), True, (1, 1024)),
+            (FakeManila(), False, (1024,)),
+            (FakeManilaV2(), True, (1, 1024)),
+            (FakeManilaV2(), False, (1024,)),
```

Comment:
I think this is fine for now if its hard to track down where its coming from, but the return shape should preferably be the same regardless of whether we are using primitives, so I would say you found a small bug here.

Reply:
> Is the discrepancy in shape only for `sample` and other measurements that go through the `execute_runtime` function, or does it happen for other execution types like `expval` or `probs` as well?

This is a good point. I didn't investigate **that** closely but I imagine it should be due to `execute_runtime` itself, so it should include other execution types like expval and probs. That's my guess though, I can write a note here to investigate further

Reply:
> I think this is fine for now if its hard to track down where its coming from, but the return shape should preferably be the same regardless of whether we are using primitives, so I would say you found a small bug here.

Yeah I think it's a small bug. Best to address in a separate PR though I figured.

Reply:
Makes sense - I'd suggest adding a shortcut story to track it, but once that's done, I'd say this is good for this PR

Reply:
[sc-62867] Added!